### PR TITLE
restore kube-cross build image and bump go to version 1.16

### PR DESCRIPTION
**Pull request by ilya-zuyev**
closes https://github.com/kubernetes/minikube/issues/10276

This PR restores BUILD_IMAGE to us.gcr.io/k8s-artifacts-prod/build-image/kube-cross:v1.16.0-1 and updates go.mod file and CI scripts to use 1.16 version

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -44,13 +44,7 @@ COMMIT ?= $(if $(shell git status --porcelain --untracked-files=no),"${COMMIT_NO
...
 HYPERKIT_BUILD_IMAGE 	?= karalabe/xgo-1.12.x
 
-# NOTE: "latest" as of 2021-02-06. kube-cross images aren't updated as often as Kubernetes
-# https://github.com/kubernetes/kubernetes/blob/master/build/build-image/cross/VERSION
-#
-# TODO: See https://github.com/kubernetes/minikube/issues/10276
-#BUILD_IMAGE 	?= us.gcr.io/k8s-artifacts-prod/build-image/kube-cross:v$(GO_VERSION)-1
-BUILD_IMAGE 	?= golang:1.16.0-buster
-#
+BUILD_IMAGE 	?= us.gcr.io/k8s-artifacts-prod/build-image/kube-cross:v$(GO_VERSION)-1
```
Comment (medyagh) [inline Makefile:47]: plz keep the comment where people can get the latest version 

# https://github.com/kubernetes/kubernetes/blob/master/build/build-image/cross/VERSION

File: go.mod [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module k8s.io/minikube
 
-go 1.15
+go 1.16
```
Comment (medyagh) [inline go.mod:3]: ensure to do go mod tidy plz

**Review by medyagh — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (ilya-zuyev) [reply Makefile:50]: fixed

File: go.mod (hunk 2 above)
Comment (ilya-zuyev) [reply go.mod:3]: done

**Comment by sharifelgamal** [comment]
ok-to-build-image

**Comment by sharifelgamal** [comment]
ok-to-build-image

**Comment by medyagh** [comment]
failed on jenkins 
https://a646c87040050000000000000000001.proxy.googleprod.com/job/Build_Cross/17003/console

File: Makefile [hunk 3]
```diff
@@ -32,7 +32,11 @@ RPM_VERSION ?= $(DEB_VERSION)
 RPM_REVISION ?= 0
 
 # used by hack/jenkins/release_build_and_upload.sh and KVM_BUILD_IMAGE, see also BUILD_IMAGE below
-GO_VERSION ?= 1.15.8
+GO_VERSION ?= 1.16.0
```
Comment (medyagh) [inline Makefile:35]: please change this version everywhere in the code (jenkins scripts and gh action things)

**Review by medyagh — CHANGES_REQUESTED** [review]

File: Makefile (hunk 3 above)
Comment (ilya-zuyev) [reply Makefile:35]: done

**Review by medyagh — APPROVED** [review]

**Pull request merged**